opm/parser/eclipse/EclipseState/Tables/JFunc.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ Copyright 2016  Statoil ASA.
+
+ This file is part of the Open Porous Media project (OPM).
+
+ OPM is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OPM is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/JFunc.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/J.hpp>
+
+namespace Opm {
+
+    JFunc::JFunc(const Deck& deck) :
+                    m_exists(deck.hasKeyword("JFUNC"))
+    {
+        if (!m_exists)
+            return;
+        const auto& kw = *deck.getKeywordList<ParserKeywords::JFUNC>()[0];
+        const auto& rec = kw.getRecord(0);
+        const auto& kw_flag = rec.getItem("FLAG").get<std::string>(0);
+        if (kw_flag == "BOTH")
+            flag = Flag::BOTH;
+        else if (kw_flag == "WATER")
+            flag = Flag::WATER;
+        else if (kw_flag == "GAS")
+            flag = Flag::GAS;
```

Comment:
in `JFUNC`, `flag` is optional and defaulted to `BOTH`.  Meaning that if it is not specified in deck, through the `JFUNC` json, it will be given as `BOTH`.

There is a potential issue here, namely if a user specifies `JFUNC GARBAGE/` in the deck.  Then we might reach a state where we read an uninitialized `flag`.

I could add a final `else throw unrecognized flag GARBAGE` if you'd like.